### PR TITLE
Add boolean support, fixes #111

### LINKED ISSUE #111: [DynamoDB]: No support for Boolean field type
It seems that [this match](https://github.com/seratch/AWScala/blob/master/src/main/scala/awscala/dynamodbv2/AttributeValue.scala#L21) is not entirely exhaustive, as it does not account for [boolean field types](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/model/AttributeValue.html#getBOOL()).

### PR DESCRIPTION
**Pull request by DeaconDesperado**
Adds support for the boolean datatype, fixing #111

### CODE REVIEW

**Comment by DeaconDesperado** [comment]
@seratch Thoughts?  I couldn't seem to eliminate the need for lines 42 and 64, since the implicit conversion to boolean throws an NPE even though it's wrapped with an option.

I have another pending branch adding nested list support and improving document maps incoming.

File: src/main/scala/awscala/dynamodbv2/AttributeValue.scala [hunk 1]
```diff
@@ -57,6 +61,9 @@ case class AttributeValue(
 ) extends aws.model.AttributeValue {
 
   setS(s.orNull[String])
+  if (bl.isDefined) {
```
Comment (seratch) [inline src/main/scala/awscala/dynamodbv2/AttributeValue.scala]: `bl.foreach(b => setBOOL(b))` is preferred.

File: src/main/scala/awscala/dynamodbv2/AttributeValue.scala [hunk 2]
```diff
@@ -37,6 +38,8 @@ object AttributeValue {
 
   def apply(v: aws.model.AttributeValue): AttributeValue = new AttributeValue(
     s = Option(v.getS),
+    //Unfortunately the ugly check is necessary here to bypass implicit conversion
+    bl = if (v.getBOOL != null) Option(v.getBOOL) else None,
```
Comment (seratch) [inline src/main/scala/awscala/dynamodbv2/AttributeValue.scala]: Simlpy `Option(v.getBOOL)` is fine here. `Option.apply` embrace null as `None`.

Comment (DeaconDesperado) [inline src/main/scala/awscala/dynamodbv2/AttributeValue.scala]: It's funny, I had assumed the same but it seems this still raises an NPE.  Will dig deeper:

```
[info]   java.lang.NullPointerException:
[info]   at scala.Predef$.Boolean2boolean(Predef.scala:361)
[info]   at awscala.dynamodbv2.AttributeValue$.apply(AttributeValue.scala:41)
[info]   at awscala.dynamodbv2.Item$$anonfun$apply$1.apply(Item.scala:9)
[info]   at awscala.dynamodbv2.Item$$anonfun$apply$1.apply(Item.scala:9)
[info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
[info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
[info]   at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
[info]   at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
[info]   at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
[info]   at scala.collection.AbstractTraversable.map(Traversable.scala:104)
```

Comment (DeaconDesperado) [inline src/main/scala/awscala/dynamodbv2/AttributeValue.scala]: I think I found the solution, you need to first unbox to a primitive bool for the implicit conversion to work.

```
bl = Option[java.lang.Boolean](v.getBOOL).map(_.booleanValue()),
```

**Comment by seratch** [comment]
Thanks!

**Pull request merged**